html/cross-origin-embedder-policy/resources/worker-owner.js

```diff
@@ -0,0 +1,35 @@
+const is_worker = !('window' in self);
+const parent_or_self = is_worker ? self : self.parent;
+
+function startWorkerAndObserveReports(worker_url, wait_for_report) {
+  const worker = new Worker(worker_url);
+  const result_promise = new Promise(resolve => {
+    worker.onmessage = _ => resolve('success');
+    worker.onerror = _ => resolve('error');
+  });
+  worker.postMessage("postMessage('reply to frame from worker');");
+
+  const report_promise = new Promise(resolve => {
+    const observer = new ReportingObserver(reports => {
+      observer.disconnect();
+      resolve(reports.map(r => r.toJSON()));
+    });
+    observer.observe();
+  });
+
+  if (wait_for_report) {
+    Promise.all([result_promise, report_promise]).then(results => {
+      parent_or_self.postMessage(results[1]);
+    });
+  } else {
+    result_promise.then(result => {
+      parent_or_self.postMessage([]);
```

Comment:
I think this should use a timeout to verify that no report is actually sent. The tests could use the new `async_promise_test` so timeouts are parallelized.

Reply:
Checking by timeout may cause the test flakiness so I'd like to avoid using timeout, however, I don't have a good idea to verify whether or not "no report" is sent without timeout.

Reply:
Changed in https://github.com/web-platform-tests/wpt/pull/28261/commits/699033558f1b5272f3c40ddab42f4833fc589509

The timeouts are in sequence because `promise_test` runs in sequence. We could use `async_test` but would need to rewrite the code to not use `await` since `async_test` needs everything to run in a `step`/`step_func`; I'd rather switch to `async_promise_test` when available.

The 2000ms timeout could be longer to reduce flakyness.

cc @stephenmcgruer FYI